packages/server-wallet/src/evm-validator.ts

```diff
@@ -44,7 +44,18 @@ export const validateTransitionWithEVM = async (
     Uint8Array.from(Buffer.from(bytecode.substr(2), 'hex')),
     Uint8Array.from(Buffer.from(data ? data.toString().substr(2) : '0x00', 'hex'))
   );
-
   // We need to ensure the result is the correct length otherwise we might be interpreting a failed assertion
-  return result.length === 32 && (utils.defaultAbiCoder.decode(['bool'], result)[0] as boolean);
+  const transitionPassed =
+    result.length === 32 && (utils.defaultAbiCoder.decode(['bool'], result)[0] as boolean);
+
+  if (!transitionPassed) {
+    // TODO: Figure out the proper encoding.
```

Comment:
This might be an easy fix but my knowledge of text encoding is pretty weak.

Reply:
Do you have a sample output from this?

Reply:
input: 
```
[8,195,121,160,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,32,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,40,67,111,117,110,116,105,110,103,65,112,112,58,32,67,111,117,110,116,101,114,32,109,117,115,116,32,98,101,32,105,110,99,114,101,109,101,110,116,101,100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
```

output:
```
\b�y�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000(CountingApp: Counter must be incremented\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000
```

Reply:
My naive attempt:
```
> b = Buffer.from(<INPUT>);
> encoder.decode(['string'], '0x' + b.toString('hex'))
Uncaught:
Error: overflow (fault="overflow", operation="toNumber", value="3963877391197344453575983046348115674221700746820753546331534351508065746944", code=NUMERIC_FAULT, version=bignumber/5.0.7)
    at Logger.makeError (/Users/andrewstewart/Code/magmo/monorepo/node_modules/@ethersproject/logger/lib/index.js:179:21)
    at Logger.throwError (/Users/andrewstewart/Code/magmo/monorepo/node_modules/@ethersproject/logger/lib/index.js:188:20)
    at throwFault (/Users/andrewstewart/Code/magmo/monorepo/node_modules/@ethersproject/bignumber/lib/bignumber.js:278:19)
    at BigNumber.toNumber (/Users/andrewstewart/Code/magmo/monorepo/node_modules/@ethersproject/bignumber/lib/bignumber.js:146:13)
    at /Users/andrewstewart/Code/magmo/monorepo/node_modules/@ethersproject/abi/lib/coders/array.js:89:60
    at Array.forEach (<anonymous>)
    at Object.unpack (/Users/andrewstewart/Code/magmo/monorepo/node_modules/@ethersproject/abi/lib/coders/array.js:85:12)
    at TupleCoder.decode (/Users/andrewstewart/Code/magmo/monorepo/node_modules/@ethersproject/abi/lib/coders/tuple.js:39:49)
    at AbiCoder.decode (/Users/andrewstewart/Code/magmo/monorepo/node_modules/@ethersproject/abi/lib/abi-coder.js:93:22) {
  reason: 'overflow',
  code: 'NUMERIC_FAULT',
  fault: 'overflow',
  operation: 'toNumber',
  value: '3963877391197344453575983046348115674221700746820753546331534351508065746944'
}
```

Reply:
Wait, when you said it's readable but garbled, it took me a long time to realize there was actual text in this line :P
> \b�y�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000(CountingApp: Counter must be incremented\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000

Reply:
Another naive attempt:
```
for (let i = 0; i < input.length; i++) {
  try {
    console.log(decoder.decode(['string'], input.slice(idx)))
  } catch (e) {
    
  }
}
```

Returns `undefined`, and logs nothing to the console :(